src/main/java/seedu/address/logic/commands/OpenCommand.java

```diff
@@ -44,29 +66,95 @@ public OpenCommand(TagName tagName) {
+            tagsToOpen = List.of(tagList.get(0));
+        } else {
+            // Open tag based on label
+            List<Tag> tagList = model.findFilteredTagList(new TagLabelEqualsKeywordPredicate(label));
+            if (tagList.isEmpty()) {
+                throw new CommandException(String.format(MESSAGE_LABEL_NOT_FOUND, label.toString()));
+            }
+            tagsToOpen = tagList;
         }
 
-        // Get tag and prepare file to be opened
-        Tag tagToOpen = tagList.get(0);
-        File file = new File(tagToOpen.getFileAddress().value);
+        // Perform open and get a set of opened tags
+        Map<Tag, String> openResults = openTags(tagsToOpen);
 
-        try {
-            // Open file using java.awt.Desktop
-            Desktop.getDesktop().open(file);
-        } catch (IOException | IllegalArgumentException e) {
-            throw new CommandException(e.getMessage(), e);
+        // Format result string
+        StringBuilder sb = new StringBuilder();
+        boolean hasError = false;
+        for (Tag tag : tagsToOpen) {
+            String message = openResults.get(tag);
+            if (message.startsWith("Error")) {
+                hasError = true;
+            }
+            sb.append(message).append("\n");
         }
-        return new CommandResult(String.format(MESSAGE_SUCCESS, tagToOpen.toString()));
+
+        String resultMessage = sb.toString();
+        // Remove last "\n"
+        if (resultMessage.endsWith("\n")) {
+            resultMessage = resultMessage.substring(0, resultMessage.length() - 1);
+        }
+
+        // Throw exception if some tags were not opened
+        if (hasError) {
+            throw new CommandException(resultMessage);
+        }
+
+        return new CommandResult(resultMessage);
+    }
+
+    /**
+     * Opens all tags from a list and returns a map of runtime messages.
+     */
+    private Map<Tag, String> openTags(List<Tag> tags) {
+        Map<Tag, String> result = new HashMap<>();
+
+        // Open every tags in the list and store error messages
+        tags.forEach(tag -> {
+            File file = new File(tag.getFileAddress().value);
+            try {
+                Desktop.getDesktop().open(file);
+                result.put(tag, String.format(MESSAGE_SUCCESS, tag));
+            } catch (IOException | IllegalArgumentException e) {
+                result.put(tag, String.format(MESSAGE_ERROR, tag, e.getMessage()));
+            }
+        });
+
+
+
+        return result;
     }
 
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof OpenCommand // instanceof handles nulls
-                && tagName.equals(((OpenCommand) other).tagName));
+        if (other == this) {
+            return true;
+        } else if (other == null) {
+            return false;
+        } else if (!(other instanceof OpenCommand)) {
```

Comment:
Is it better to combine the if check to one using "or" operator since both of them return false?

Reply:
I think writing `equals()` this way is very clear and easy to understand, like a pipeline.

Reply:
Okay!